app/controllers/users_controller.rb

```diff
@@ -130,7 +138,56 @@ def activity
     render layout: 'without_sidebar'
   end
 
-  def mod; end
+  def mod
+    render layout: 'without_sidebar'
+  end
+
+  def mod_escalation
+    @flag = Flag.new(post_flag_type: nil, reason: '', post_id: @user.id, post_type: 'User', user: current_user)
+    render layout: 'without_sidebar'
+  end
+
+  def mod_escalate
+    @flag = Flag.create(post_flag_type: nil, reason: params[:flag][:reason], post_id: @user.id,
+                        post_type: 'User', user: current_user, escalated: true,
+                        escalated_by: current_user, escalated_at: DateTime.now,
+                        escalation_comment: '(escalated via Contact Community Team Tool)')
+    FlagMailer.with(flag: @flag).flag_escalated.deliver_now
+    flash[:success] = 'Thank you for your message. We have been notified and are looking into it.'
+    redirect_to mod_user_path(@user)
+  end
+
+  def mod_contact
+    render layout: 'without_sidebar'
+  end
+
+  def mod_message
+    title = params[:title]
+    unless title.present?
+      title = 'Private Moderator Message'
+    end
+
+    body = params[:body]
+
+    @comment_thread = CommentThread.new(title: title, post: nil, is_private: true)
+    @comment = Comment.new(post: nil, content: body, user: current_user, comment_thread: @comment_thread)
+
+    success = ActiveRecord::Base.transaction do
+      @comment_thread.save!
+      @comment.save!
```

Comment:
When running this code, by selecting "Contact Privately" from the "Moderator Tools" tab of a user profile, and then sending a message, an error is displayed "Validation failed: Post must exist".

Reply:
Now done in #767